### ISSUE TITLE
{Admin] Rename 'ship order' to 'order picked up', when order uses a pickup method

### ISSUE THREAD

**Issue by drummer83** [openfoodfoundation/wishlist#287]
## What is the need / problem?
The platform supports pickup methods and delivery methods for orders. Since [#9682](https://github.com/openfoodfoundation/openfoodnetwork/pull/9682) we also have separate email notifications for those two. To have a consistent user experience in the admin area we should rename the buttons to mark an order as shipped to 'picked up' in case the order has a pickup method selected.

## Which type of users does this problem affect (and how many, if known)?
- Hub Managers (MCFEs)

## Potential solutions that will solve the problem
- Adjust the following locations (if a pickup method is selected for the order):
  - ship icon in orders list (/admin/orders)
    - use a little bag as icon
    - adjust the tooltip text to 'picked up'
    - ![image](https://user-images.githubusercontent.com/1790176/194717662-17d47ed1-6142-4fb2-be6d-d30852893b9d.png)
  - ship button on edit order page (/admin/orders/R...../edit)
    - adjust the text to 'picked up'
    - ![image](https://user-images.githubusercontent.com/1790176/194717732-eb20ac9c-6e35-45c5-b911-4fdb836e5a36.png)
  - ship order command in the actions menu
    - use a little bag as icon
    - adjust text to 'order picked up'
    - ![image](https://user-images.githubusercontent.com/1790176/194717768-80ff5fc2-8b4e-4c30-8b37-c39b319b1e02.png)

**Comment by tschumilas** [comment]
Question about this --  at present when a store manager selects the 'shipped' icon,  the shipment state changes from 'ready' to 'shipped'.  This effectively 'locks' the order.   Products on 'shipped' orders no longer show on a Bulk Order Management search for example.

Would this functionality be the same if there was a 'picked up' state?

I'm not arguing for or against this - just want to be clear.     Here in OFN-CAN -  payment methods are almost always 'cash'  or bank transfer (which in the OFN app  is a manual method like cash).  So users do not mark orders as shipped now because they often need to search details of the order up until payment is made.  And often there are returns and refunds AFTER pickup - so the shop manager needs to keep orders open for a while for that.  .  So most of our users can't use 'shipped'  or a future 'picked up'  if it 'locks' orders.  Its a pity because users have trouble keeping track of who has picked up and who hasn't.  In my own shop,  I use a separate spreadsheet outside the app to track who has picked up orders.   The app does a great job of helping me track who has paid -   but the only way to track who has received their order is the shipped state -  which makes it impossible to search order contents.

**Comment by audez** [comment]
Thanks for this one @drummer83 . Would it be possible to add to the issue to change the "Shipment state" column text to "Picked up" instead of "Shipped"?
![Screen Shot 2022-11-21 at 12 27 35](https://user-images.githubusercontent.com/28386648/203039643-5ec6ca31-3ed2-4529-adb7-e4c0d5b36ee5.png)

@tschumilas thanks for this input..So the idea would be that when a "pickup" order shipped state is set to "picked up", we keep the same behavior as when a "pickup" order is set to "ready"? (orders can still be edited)
Actually wouldn't that be interesting for shipped orders too? I mean if a producer ships an order, and for instance has forgotten a product, he's not able to adjust the order anymore. 

Or maybe we need a further final step? 
"pending" -> "ready" -> "shipped"/"picked up" -> "archive" (order is locked and disappears from BOM - and it would be nice that in "Orders" page it would "disappear" too or go below all active orders, which was actually a request from one of our hubs)
Plus if we uncorrelate the shipment and payment status, shop managers would easily be able to "unarchive" the orders if they need to make adjustments.

**Comment by filipefurtad0** [comment]
To add to the discussion today on the papercut-meeting @RachL @jibees and to refresh my mind -  yes, we can ship orders with pick-up shipping methods:

![image](https://user-images.githubusercontent.com/49817236/206184235-f6fd8a10-e616-4d62-a58b-f0245ba13de5.png)

In terms of data-model:
- `shipment_state` property of the `spree_orders`, and is independent of the shipment method type
- orders currently must have a `spree_shipments` associated (like `H45613165735` above)
- when we "ship" then `shipment_state` becomes `shipped`

So, perhaps the way to make this work as a papercut would be to keep the data model as is, and change only the UI as a function of whether a shipping method is type delivery or pick-up (i.e., `require_ship_address` = `true` or `false`, respectively).

This does seems straightforward to cover in a system spec/manual test, so it would be a papercut from the testing side of it.

Does it sound doable @jibees from a dev perspective?

A change in the data model sounds like a big overhead as it would probably mean:
- creating another `shipment_state` like `picked_up` or `received`, for pick up shipping methods
- and/or to be able to create orders without any `spree_shipments` associated

**Comment by jibees** [comment]
> Does it sound doable @jibees from a dev perspective?

Yes. As there is no change and no downside from the data model point of view, only improvements in term of UX, i think it's a good idea. We can still improve the data model later on.
